### PR TITLE
Onboard command extension for Azure Managed Grafana service

### PR DESCRIPTION
**Pull request by yugangw-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Azure Managed Grafana service

**Comment by yonzhan** [comment]
@yugangw-msft please let us know when this PR is ready to review/merge from CLI side.

File: src/amg/azext_amg/commands.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# pylint: disable=line-too-long
+
+from ._validators import process_grafana_create_namespace
+
+
+def load_command_table(self, _):
+
+    with self.command_group('grafana', is_preview=True) as g:
+        g.custom_command('create', 'create_grafana', validator=process_grafana_create_namespace)
+        g.custom_command('delete', 'delete_grafana')
```
Comment (kairu-ms) [inline src/amg/azext_amg/commands.py:15]: You can use `confirmation=True` which will add `--yes` argument by azure-cli-core framework and prop user confirmation when run delete command.
 
https://github.com/Azure/azure-cli-extensions/blob/943d81d738ee3bc6283b59fba98860338fb2da98/src/network-manager/azext_network_manager/commands.py#L72

File: src/amg/azext_amg/_params.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                               "numeric value and is only unique per Grafana install."))
+        c.argument("folder", help="id, uid, title which can identify a folder. CLI will search in the order of id, uid, and title, till finds a match")
+
+    with self.argument_context("grafana create") as c:
+        c.argument("grafana_name", grafana_name_type, options_list=["--name", "-n"], validator=None)
+        c.argument("skip_system_assigned_identity", options_list=["-s", "--skip-system-assigned-identity"], arg_type=get_three_state_flag(), help="Do not enable system assigned identity")
+        c.argument("skip_role_assignments", arg_type=get_three_state_flag(), help="Do not create role assignments for managed identity and the current login user")
+
+    with self.argument_context("grafana delete") as c:
+        c.argument('yes', options_list=['--yes', '-y'], help='Do not prompt for confirmation.', action='store_true')
```
Comment (kairu-ms) [inline src/amg/azext_amg/_params.py:33]: When register commands with `confirmation=True` this argument will not need to state.

File: src/amg/azext_amg/_validators.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+def process_missing_resource_group_parameter(cmd, namespace):
+    if not namespace.resource_group_name and namespace.grafana_name:
+        client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
+        resources = client.resources.list(filter="resourceType eq 'Microsoft.Dashboard/grafana'")
+        resources = list(resources)
+        match = next((i for i in resources if i.name == namespace.grafana_name), None)
+        if match:
+            namespace.resource_group_name = parse_resource_id(match.id)["resource_group"]
+        else:
+            raise CLIError((f"Not able to find the Grafana instance: '{namespace.grafana_name}'. Please "
```
Comment (kairu-ms) [inline src/amg/azext_amg/_validators.py:30]: CLIError is not recommended to raise, Please use a specific error by following this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

**Comment by yugangw-msft** [comment]
@yonzhan it is ready for review CC @kairu-ms

**Comment by yonzhan** [comment]
Add @jiasli to help review this PR since there are a bunch of auth and service principle related code

File: src/amg/azext_amg/_params.py (hunk 2 above)
Comment (yugangw-msft) [reply src/amg/azext_amg/_params.py]: updated

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**